src/link/edit.js

```diff
@@ -24,6 +26,11 @@ const Edit = ( {
 	const [ addingPreview, setAddingPreview ] = useState( false );
 	const startAddingPreview = () => setAddingPreview( true );
 	const stopAddingPreview = () => setAddingPreview( false );
+	const [ viewingPreview, setViewingPreview ] = useState( false );
+	const startViewingPreview = () => setViewingPreview( true );
+	const stopViewingPreview = () => setViewingPreview( false );
+	const [ previewTitle, setPreviewTitle ] = useState();
+	const activePreview = getActiveFormat( value, name );
```

Comment:
Looking into this and see if we can simplify this

Reply:
that would be great

Reply:
Ok, I have done the simplify and some bug fixes, please review and test it carefully.